tests/integration/server.py

```diff
@@ -0,0 +1,206 @@
+        logging.error('Error while writing custom metric: {0}'.format(e))
+        raise ErrorResponse('Error while writing custom metric: {0}'.format(e))
+
+    return 'OK', 200
+
+
+def _write_metric(name, client, token):
+    """
+    Write a metric to Stackdriver Monitoring.
+
+    Keyword arguments:
+    name -- The name of the metric to write. Takes the form
+    'custom.googleapis.com/{metric_name}'
+    client -- the authenticated instance of a Google Cloud Client
+    token -- a random 64-bit integer token. The metric value to be written.
+
+    Once the metric is written, the test driver will retrieve all metrics
+    written with the provided name, and verify there is an entry with the
+    same value as the provided token.
+    """
+    metric = client.metric(name, {})
+    resource = client.resource('global', labels={})
+    client.write_point(metric, resource, token)
+
+
+def _create_descriptor(name, client):
+    """
+    Create a new metric descriptor.
+    This descriptor is implicitly used to write a point-value metric to
+    Stackdriver.
+    """
+    logging.info('No descriptor found with name {0}: Creating...'.format(name))
+    descriptor = client.metric_descriptor(
+        name,
+        metric_kind=google.cloud.monitoring.MetricKind.GAUGE,
+        value_type=google.cloud.monitoring.ValueType.INT64,
+        description='Test Metric'
+        )
+    descriptor.create()
+
+
+@app.route('/exception', methods=['POST'])
+def _exception():
+    request_data = request.get_json()
+    if request_data is None:
+        raise ErrorResponse('Unable to parse request JSON: '
+                            'did you set the Content-type header?')
+    token = request_data.get('token')
+    if not token:
+        raise ErrorResponse('Please provide token')
+
+    try:
+        client = google.cloud.error_reporting.Client()
+        try:
+            raise NameError
+        except Exception:
+            client.report_exception()
+
+        client.report(token)
+    except Exception as e:
+        logging.error('Error while reporting exception: {0}'.format(e))
+        raise ErrorResponse('Error while reporting exception: {0}'.format(e))
+
+    return 'OK', 200
+
+
+@app.route('/trace', methods=['POST'])
+def _trace():
+    return 'OK', 204
+
+
+class ErrorResponse(Exception):
+    status_code = 400
+
+    def __init__(self, message, status_code=None, payload=None):
+        Exception.__init__(self)
+        self.message = message
+        if status_code is not None:
+            self.status_code = status_code
+        self.payload = payload
+
+    def to_dict(self):
+        rv = dict(self.payload or ())
+        rv['message'] = self.message
+        return rv
+
+
+@app.errorhandler(ErrorResponse)
+def handle_invalid_usage(error):
+    response = jsonify(error.to_dict())
+    response.status_code = error.status_code
+    return response
+
+
+if __name__ == '__main__':
+    app.run(debug=True, host='0.0.0.0', port=8080)
```

Comment:
ahh ok that makes sense. first time I've ever deployed with gunicorn so I'm still learning :)